### PR TITLE
[SIMPLY-2858] Fix ordering of testing libraries, when enabled.

### PR DESCRIPTION
**Pull request by twaddington**
**What's this do?**
The list of testing libraries already includes the production libraries, so only load one or the other. This means the testing libraries are no longer just appended to the end of the list (making them difficult to find).

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SIMPLY-2858

**How should this be tested? / Do these changes have associated tests?**
Make sure enabling testing libraries in debug settings works as expected.

**Dependencies for merging? Releasing to production?**
n/a

**Has the application documentation been updated for these changes?**
n/a

**Did someone actually run this code to verify it works?**
@twaddington ran the Vanilla app.

### CODE REVIEW

File: simplified-accounts-registry/src/main/java/org/nypl/simplified/accounts/registry/AccountProviderRegistry.kt [hunk 1]
```diff
@@ -102,6 +102,8 @@ class AccountProviderRegistry private constructor(
   }
 
   override fun clear() {
+    this.descriptions.clear()
```
Comment (twaddington) [inline simplified-accounts-registry/src/main/java/org/nypl/simplified/accounts/registry/AccountProviderRegistry.kt:105]: We only call clear in one place, so this shouldn't negatively affect anything.

File: simplified-accounts-source-nyplregistry/src/main/java/org/nypl/simplified/accounts/source/nyplregistry/AccountProviderSourceNYPLRegistry.kt [hunk 2]
```diff
@@ -244,18 +245,17 @@ class AccountProviderSourceNYPLRegistry(
...
-            this.logger.debug("{} cached production providers", countProduction)
-            this.logger.debug("{} cached testing providers", countTesting)
-            result.result.providers.associateBy(AccountProviderDescription::id).toMap()
+            this.logger.debug("{} cached providers", result.result.providers.size)
+            result.result.providers.associateBy(AccountProviderDescription::id)
           }
         }
       }
+    } catch (e: FileNotFoundException) {
+      this.logger.debug("no cache file exists, skipping")
```
Comment (twaddington) [inline simplified-accounts-source-nyplregistry/src/main/java/org/nypl/simplified/accounts/source/nyplregistry/AccountProviderSourceNYPLRegistry.kt:254]: Avoid spamming logcat with an exception for an expected error.

File: simplified-accounts-registry/src/main/java/org/nypl/simplified/accounts/registry/AccountProviderRegistry.kt [hunk 3]
```diff
@@ -102,6 +102,8 @@ class AccountProviderRegistry private constructor(
   }
 
   override fun clear() {
```
Comment (twaddington) [inline simplified-accounts-registry/src/main/java/org/nypl/simplified/accounts/registry/AccountProviderRegistry.kt:104]: We only call clear in one place, so this change shouldn't affect anything unexpected.

**Review by io7m — APPROVED** [review]
I like it!

**Comment by twaddington** [comment]
I think the build got stuck so I'm gonna merge.

**Comment by io7m** [comment]
It got there eventually.

**Pull request merged**